src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,11 +1,33 @@
+import React, { Dispatch, SetStateAction, useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  onSelectedTodo: Dispatch<SetStateAction<Todo | null>>;
+  selectedUserId: number;
+  selectedTodo: Todo | null;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  onSelectedTodo,
+  selectedUserId,
+  selectedTodo,
+}) => {
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    getUser(selectedUserId).then(setUser);
```

Comment:
This is a minor issue, but it's a good practice to handle errors in your promises. If `getUser` fails for some reason, `setUser` will be called with an undefined value. Consider adding a `.catch` block to handle any errors.